shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts

```diff
@@ -249,14 +265,182 @@ export class BaseCore implements IAppInsightsCore {
             };
 
             _self.releaseQueue = () => {
-                if (_eventQueue.length > 0) {
-                    arrForEach(_eventQueue, (event: ITelemetryItem) => {
+                if (_isInitialized && _eventQueue.length > 0) {
+                    let eventQueue = _eventQueue;
+                    _eventQueue = [];
+
+                    arrForEach(eventQueue, (event: ITelemetryItem) => {
                         _self.getProcessTelContext().processNext(event);
                     });
-
-                    _eventQueue = [];
                 }
             };
+
+            /**
+             * Periodically check logger.queue for log messages to be flushed
+             */
+             _self.pollInternalLogs = (eventName?: string): number => {
+                 _internalLogsEventName = eventName;
+
+                let interval = _self.config.diagnosticLogInterval;
+                if (!interval || !(interval > 0)) {
+                    interval = 10000;
+                }
+                if(_internalLogPoller) {
+                    clearInterval(_internalLogPoller);
+                }
+                _internalLogPoller = setInterval(() => {
+                    _flushInternalLogs();
+                }, interval) as any;
+
+                return _internalLogPoller;
+            }
+
+            /**
+             * Stop polling log messages from logger.queue
+             */
+            _self.stopPollingInternalLogs = (): void => {
+                if(_internalLogPoller) {
+                    clearInterval(_internalLogPoller);
+                    _internalLogPoller = 0;
+                    _flushInternalLogs();
+                }
+            }
+
+            // Add addTelemetryInitializer
+            proxyFunctions(_self, () => _telemetryInitializerPlugin, [ "addTelemetryInitializer" ]);
+
+            _self.getPlugin = _getPlugin;
+
+            function _initDefaults() {
+                _isInitialized = false;
+
+                // Use a default logger so initialization errors are not dropped on the floor with full logging
+                _self.logger = new DiagnosticLogger({ loggingLevelConsole: LoggingSeverity.CRITICAL });
+                _self.config = null;
+                _self._extensions = [];
+
+                _telemetryInitializerPlugin = new TelemetryInitializerPlugin();
+                _eventQueue = [];
+                _notificationManager = null;
+                _perfManager = null;
+                _cfgPerfManager = null;
+                _cookieManager = null;
+                _pluginChain = null;
+                _coreExtensions = null;
+                _configExtensions = null;
+                _channelControl = null;
+                _channelConfig = null;
+                _channelQueue = null;
+                _internalLogsEventName = null;
+            }
+
+            // Initialize or Re-initialize the plugins
+            function _initPluginChain(config: IConfiguration) {
```

Comment:
One of the big changes in this PR is that the PluginChain can now be created once and reused for every event, previously a new context and chain were created for every event. - That was a long standing performance bug.

The intent of the "context" is so that there is one for each event so as each event is processed through the system it can handle both the old (prior to the context `setNextPlugin()` only plugins) and the newer `ctx.processNext()` so that the SDK can keep track of and update any required context.
Such as the upcoming the order of execution completely changing during async (add/remove plugins) operations in the same running plugin and SDK instance.